packages/superset-ui-core/src/query/types/PostProcessing.ts

```diff
@@ -115,4 +160,12 @@ export type PostProcessingRule =
   | PostProcessingBoxplot
   | PostProcessingContribution
   | PostProcessingPivot
-  | PostProcessingProphet;
+  | PostProcessingProphet
+  | PostProcessingDiff
+  | PostProcessingRolling
+  | PostProcessingCum
+  | PostProcessingCompare;
+
+export interface PostProcessingFactory<T> {
+  (formData: QueryFormData, queryObject: QueryObject): T;
+}
```

Comment:
Since this operator building function may only be used with chart controls, maybe we should move it to the `chart-controls` package.

Reply:
Nice! Thank you.